examples/Chat/Example06_ChatSerialization.cs

```diff
@@ -0,0 +1,104 @@
+﻿using NUnit.Framework;
+using OpenAI.Chat;
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace OpenAI.Examples;
+
+public partial class ChatExamples
+{
+    #region
+    public static List<ChatMessage> DeserializeMessages(BinaryData data)
+    {
+        List<ChatMessage> messages = [];
+        using JsonDocument messagesAsJson = JsonDocument.Parse(data.ToString());
+
+        foreach (JsonElement jsonElement in messagesAsJson.RootElement.EnumerateArray())
+        {
+            BinaryData jsonElementAsData = BinaryData.FromString(jsonElement.ToString());
```

Comment:
Once [this ships](https://github.com/dotnet/runtime/issues/77666), I believe we should be able to improve perf here by not round-tripping UTF-8 through UTF-16 with strings ...